lib/StripeResource.js

```diff
@@ -171,6 +173,14 @@ StripeResource.prototype = {
             null
           );
         }
+
+        if (self._stripe.getTelemetryEnabled() && res.requestId) {
+          self._stripe._prevRequestMetrics.push({
+            'request_id': res.requestId,
+            'request_duration_ms': requestDurationMs,
+          });
+        }
```

Comment:
Like for stripe-python, can we put these into a `_addTelemetryHeader` and `_recordRequestMetrics` or such?

Reply:
This unbounded buffer worries me a little bit, do you think that we could have a check against some arbitrarily large size (e.g. `> 100`) and if we get there, don't push on a new value and then use `utils.emitWarning` to indicate that we dropped a request result? (So that we can tell it's happening.)